aptos-move/aptos-vm/src/natives.rs

```diff
@@ -35,11 +35,14 @@ pub fn aptos_natives(
     timed_features: TimedFeatures,
     features: Arc<Features>,
 ) -> NativeFunctionTable {
+    let sha256_gas_params: move_stdlib::natives::hash::Sha2_256GasParameters =
+        gas_params.move_stdlib.hash.sha2_256.clone();
     move_stdlib::natives::all_natives(CORE_CODE_ADDRESS, gas_params.move_stdlib)
         .into_iter()
         .filter(|(_, name, _, _)| name.as_str() != "vector")
         .chain(aptos_framework::natives::all_natives(
             CORE_CODE_ADDRESS,
+            sha256_gas_params,
```

Comment:
You could actually pass down the full gas_params here, since in the future other natives might need them.

Reply:
Naively passing `gas_params` to aptos_framework leads to circular dependencies between aptos_gas and aptos_framework...
But what's currently implemented is also not ideal: aptos_framework depending on move_stdlib is probably not desired.

I think it's a separate refactoring work to make it easier to share gas parameters across aptos-stdlib and move-stdlib, and is beyond the scope of this change. @wrwg your thoughts?

Reply:
I don't understand what you mean. I can make the change and both `framework` and `aptos-vm` compile fine w/o any circular dependency issues. Here's a diff
```
diff --git a/aptos-move/aptos-vm/src/natives.rs b/aptos-move/aptos-vm/src/natives.rs
index 30000e63e1..6c764b9d40 100644
--- a/aptos-move/aptos-vm/src/natives.rs
+++ b/aptos-move/aptos-vm/src/natives.rs
@@ -35,14 +35,12 @@ pub fn aptos_natives(
     timed_features: TimedFeatures,
     features: Arc<Features>,
 ) -> NativeFunctionTable {
-    let sha256_gas_params: move_stdlib::natives::hash::Sha2_256GasParameters =
-        gas_params.move_stdlib.hash.sha2_256.clone();
-    move_stdlib::natives::all_natives(CORE_CODE_ADDRESS, gas_params.move_stdlib)
+    move_stdlib::natives::all_natives(CORE_CODE_ADDRESS, gas_params.move_stdlib.clone())
         .into_iter()
         .filter(|(_, name, _, _)| name.as_str() != "vector")
         .chain(aptos_framework::natives::all_natives(
             CORE_CODE_ADDRESS,
-            sha256_gas_params,
+            gas_params.move_stdlib,
             gas_params.aptos_framework,
             timed_features,
             features,
diff --git a/aptos-move/framework/src/natives/cryptography/algebra/mod.rs b/aptos-move/framework/src/natives/cryptography/algebra/mod.rs
index 75dd50c589..2d00d56c43 100644
--- a/aptos-move/framework/src/natives/cryptography/algebra/mod.rs
+++ b/aptos-move/framework/src/natives/cryptography/algebra/mod.rs
@@ -247,7 +247,7 @@ static BLS12381_Q12_LENDIAN: Lazy<Vec<u8>> = Lazy::new(|| {

 pub fn make_all(
     gas_params: crate::natives::cryptography::algebra::gas::GasParameters,
-    sha256_gas_params: move_stdlib::natives::hash::Sha2_256GasParameters,
+    move_gas_params: move_stdlib::natives::GasParameters,
     timed_features: TimedFeatures,
     features: Arc<Features>,
 ) -> impl Iterator<Item = (String, NativeFunction)> {
@@ -413,7 +413,7 @@ pub fn make_all(
             make_safe_native(
                 HashToGasParameters {
                     algebra: gas_params.clone(),
-                    sha2: sha256_gas_params,
+                    sha2: move_gas_params.hash.sha2_256,
                 },
                 timed_features.clone(),
                 features.clone(),
diff --git a/aptos-move/framework/src/natives/mod.rs b/aptos-move/framework/src/natives/mod.rs
index 0a76b3b88f..0af42f7065 100644
--- a/aptos-move/framework/src/natives/mod.rs
+++ b/aptos-move/framework/src/natives/mod.rs
@@ -264,7 +264,7 @@ impl GasParameters {

 pub fn all_natives(
     framework_addr: AccountAddress,
-    sha256_gas_params: move_stdlib::natives::hash::Sha2_256GasParameters,
+    move_gas_params: move_stdlib::natives::GasParameters,
     gas_params: GasParameters,
     timed_features: TimedFeatures,
     features: Arc<Features>,
@@ -308,7 +308,7 @@ pub fn all_natives(
         "crypt_algebra",
         cryptography::algebra::make_all(
             gas_params.algebra.clone(),
-            sha256_gas_params,
+            move_gas_params,
             timed_features.clone(),
             features.clone()
         )
```

Reply:
Oh you meant to break the top-level `gas_params` up and pass in as multiple parameters... sure, it's slightly better i think.

Reply:
on a side node,
```
size_of::<aptos_gas::NativeGasParameters>()==1480
size_of::<move_stdlib::natives::GasParameters>()==224
size_of::<move_stdlib::natives::hash::Sha2_256GasParameters>()==24
size_of::<aptos_framework::natives::cryptography::algebra::gas::GasParameters>()==488
```

Reply:
> Oh you meant to break the top-level `gas_params` up and pass in as multiple parameters... sure, it's slightly better i think.

Uhm, not sure what you mean.

I just mean, let's pass `gas_params.move_stdlib` into `aptos_framework::natives::all_natives` as per the diff above.